.github/scripts/query_observe.py

```diff
@@ -0,0 +1,299 @@
+            logger.debug("Request for query {} successful with status code {}:".format(query, response.status_code))
+            logger.debug("Response:{}".format(json_list))
+            return json_list
+    except requests.exceptions.HTTPError as err:
+        logging.debug(err.request.url)
+        logging.debug(err)
+        logging.debug(err.response.text)
+        return None
+
+
+def query_dataset(bearer_token: str, dataset_id: str, pipeline: str = "", interval: str = "30m") -> list:
+    """
+
+    Queries the last 30 minutes (default) of a dataset returning result of query. Uses Observe OpenAPI
+
+    @param bearer_token: bearer token for authorization
+    @param dataset_id: dataset_id to query using openAPI query
+    @param pipeline: OPAL Pipeline
+    @param interval: interval to query dataset, eg:"30m"
+
+    @return: dataset: queried dataset  in json separated by timestamps
+
+    See  https://developer.observeinc.com/#/paths/~1v1~1meta~1export~1query/post
+    """
+
+    logger.info("Querying Dataset for Dataset ID: {} for interval {}".format(dataset_id, interval))
+    query = """
+     {
+        "query": {
+            "stages":[
+              {
+                 "input":[
+                     {
+                     "inputName": "default",
+                     "datasetId": "%s"
+                    }
+                ],
+                "stageID":"main",
+                "pipeline": "%s"
+            }
+        ]
+      }
+    }
+    """ % (dataset_id, pipeline)
+    params = {
+        "interval": interval
+    }
+    dataset = send_query(bearer_token, query, params, url_extension='/export/query', type='openapi')
+    return dataset
+
+
+def validate_azure_data(source: str, stale_checks_mins: int = 30, query_interval: str = "30m") -> bool:
+    """
+
+    :param source: can be 'EventHub`, `ResourceManagement`, `VMMetrics`
+    :param stale_checks_mins: how long should difference be between received data and current timestamp
+    :param query_interval: default interval to valid from (last 30m default)
+    :return: if source is validated from current time (true, false)
+    """
+
+    AZURE_DATASET_ID = os.environ.get("AZURE_DATASET_ID")
+    OBSERVE_TOKEN_ID = os.environ.get("OBSERVE_TOKEN_ID")
+    bearer_token = get_bearer_token()
+
+    current_ts = datetime.datetime.now().timestamp()
+    current_ts_ns = int(current_ts * 1e9)
+    current_ts_string = datetime.datetime.fromtimestamp(current_ts, timezone.utc).strftime('%Y-%m-%dT%H:%M:%SZ')
+ 
+    logger.info("{}: stale_check_mins is: {} mins".format(source, stale_checks_mins))
+    logger.info("{}: query interval is: {}".format(source, query_interval))
+    logger.info("{}: Current time is: {}".format(source, current_ts_string))
+
+    pipeline = ''
+    if source == "EventHub":
+        # OPAL PIPELINE to execute on Dataset
+        pipeline = "make_col _ob_datastream_token_id:coalesce(DATASTREAM_TOKEN_ID, string(EXTRA.datastream_token_id))| " \
```

Comment:
I like to store queries in their own files and import - not required but consider

Reply:
Done! thx